web/src/settings_users.ts

```diff
@@ -0,0 +1,736 @@
+    });
+    $("#admin_deactivated_users_table").hide();
+    $("#admin_users_table").hide();
+
+    populate_users();
+}
+
+function handle_deactivation($tbody: JQuery): void {
+    $tbody.on("click", ".deactivate", (e) => {
+        // This click event must not get propagated to parent container otherwise the modal
+        // will not show up because of a call to `close_active` in `settings.js`.
+        e.preventDefault();
+        e.stopPropagation();
+
+        const $row = $(e.target).closest(".user_row");
+        const user_id = Number($row.attr("data-user-id"));
+
+        let url = "/json/users/" + encodeURIComponent(user_id);
+        if (user_id === current_user.user_id) {
+            url = "/json/users/me";
+        }
+
+        function handle_confirm(): void {
+            let data = {};
+            if ($(".send_email").is(":checked")) {
+                data = {
+                    deactivation_notification_comment: $(".email_field_textarea").val(),
+                };
+            }
+
+            const opts: {
+                success_continuation?: () => void;
+            } = {};
+            if (user_id === current_user.user_id) {
+                opts.success_continuation = () => {
+                    window.location.href = "/login/";
+                };
+            }
+
+            dialog_widget.submit_api_request(channel.del, url, data, opts);
+        }
+
+        user_deactivation_ui.confirm_deactivation(user_id, handle_confirm, true);
+    });
+}
+
+function handle_bot_deactivation($tbody: JQuery): void {
+    $tbody.on("click", ".deactivate", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        const $button_elem = $(e.target);
+        const $row = $button_elem.closest(".user_row");
+        const bot_id = Number.parseInt($row.attr("data-user-id")!, 10);
+
+        function handle_confirm(): void {
+            const url = "/json/bots/" + encodeURIComponent(bot_id);
+            dialog_widget.submit_api_request(channel.del, url, {});
+        }
+
+        user_deactivation_ui.confirm_bot_deactivation(bot_id, handle_confirm, true);
+    });
+}
+
+function handle_reactivation($tbody: JQuery): void {
+    $tbody.on("click", ".reactivate", (e) => {
+        e.preventDefault();
+        e.stopPropagation();
+
+        // Go up the tree until we find the user row, then grab the email element
+        const $button_elem = $(e.target);
+        const $row = $button_elem.closest(".user_row");
+        const user_id = Number.parseInt($row.attr("data-user-id")!, 10);
+
+        function handle_confirm(): void {
+            const url = "/json/users/" + encodeURIComponent(user_id) + "/reactivate";
+            dialog_widget.submit_api_request(channel.post, url, {});
+        }
+
+        user_deactivation_ui.confirm_reactivation(user_id, handle_confirm, true);
+    });
+}
+
+function handle_edit_form($tbody: JQuery): void {
+    $tbody.on("click", ".open-user-form", function (this, e) {
```

Comment:
```suggestion
    $tbody.on("click", ".open-user-form", function (e) {
```

An explicit `this` argument is only needed if we’re explicitly annotating it.